lib/smart_proxy_reports/ansible_processor.rb

```diff
@@ -71,14 +71,22 @@ def spool_report
       payload = measure :format do
         report_hash.to_json
       end
-      SpooledHttpClient.instance.spool(report_id, payload)
+      SpooledHttpClient.instance.spool("reports", report_id, payload)
     end
 
     private
 
     def process_facts(result)
       # TODO: add fact processing and sending to the fact endpoint
-      result["result"]["ansible_facts"] = {}
+      payload = {
+        "name": @body["host"],
+        "facts": {
+            "ansible_facts": result["result"]["ansible_facts"],
+            "_type": "ansible",
+            "_timestamp": get_now()
+        },
+      }
+      SpooledHttpClient.instance.spool("facts", report_id, payload)
```

Comment:
If this translates down to a filename, then probably singular would be better: `fact-XXXXXXX` and the same above: `report-XXXXXXXXX`.